python/src/aiconfig/editor/client/src/components/prompt/PromptActionBar.tsx

```diff
@@ -44,48 +45,60 @@ export default memo(function PromptActionBar({
   const promptMetadataSchema = promptSchema?.prompt_metadata;
 
   return (
-    <Flex direction="column" justify="space-between">
+    <Flex direction="column" justify="space-between" h="100%">
       {isExpanded ? (
-        <Container miw="400px">
-          <ActionIcon size="sm" onClick={() => setIsExpanded(false)}>
-            <IconClearAll />
-          </ActionIcon>
-          <Tabs defaultValue="settings">
-            <Tabs.List>
-              <Tabs.Tab value="settings">Settings</Tabs.Tab>
-              {checkParametersSupported(prompt) && (
-                <Tabs.Tab value="parameters">
-                  Local Variables (Parameters)
-                </Tabs.Tab>
-              )}
-            </Tabs.List>
+        <>
+          <Container miw="400px">
+            <ActionIcon
+              size="sm"
+              onClick={() => setIsExpanded(false)}
+              mt="0.5em"
+            >
+              <IconClearAll />
+            </ActionIcon>
+            <Tabs defaultValue="settings" mb="1em">
+              <Tabs.List>
+                <Tabs.Tab value="settings">Settings</Tabs.Tab>
+                {checkParametersSupported(prompt) && (
+                  <Tabs.Tab value="parameters">
+                    Local Variables (Parameters)
+                  </Tabs.Tab>
+                )}
+              </Tabs.List>
 
-            <Tabs.Panel value="settings">
-              <ModelSettingsRenderer
-                settings={getModelSettings(prompt)}
-                schema={modelSettingsSchema}
-                onUpdateModelSettings={onUpdateModelSettings}
-              />
-              <PromptMetadataRenderer
-                prompt={prompt}
-                schema={promptMetadataSchema}
-              />
-            </Tabs.Panel>
-
-            {checkParametersSupported(prompt) && (
-              <Tabs.Panel value="parameters">
-                <ParametersRenderer
-                  initialValue={getPromptParameters(prompt)}
-                  onUpdateParameters={onUpdateParameters}
+              <Tabs.Panel value="settings">
+                <ModelSettingsRenderer
+                  settings={getModelSettings(prompt)}
+                  schema={modelSettingsSchema}
+                  onUpdateModelSettings={onUpdateModelSettings}
+                />
+                <PromptMetadataRenderer
+                  prompt={prompt}
+                  schema={promptMetadataSchema}
                 />
               </Tabs.Panel>
-            )}
-          </Tabs>{" "}
-        </Container>
+
+              {checkParametersSupported(prompt) && (
+                <Tabs.Panel value="parameters">
+                  <ParametersRenderer
+                    initialValue={getPromptParameters(prompt)}
+                    onUpdateParameters={onUpdateParameters}
+                  />
+                </Tabs.Panel>
+              )}
+            </Tabs>
+          </Container>
+          <RunPromptButton prompt={prompt} size="full" />
```

Comment:
Nit: I'd prefer to have this in the top right of the prompt-input area (not bottom of the settings), similar to what we have in the web app. Not a blocker for MVP, just commenting

Reply:
Ya, the current placement is a bit far from the input especially for large inputs/outputs. I was thinking it might be better to have it directly across from bottom-right of the input when the action bar is collapsed, otherwise top right of the settings. In any case, this is currently matching designs in https://www.figma.com/file/7YVa6KpEDgptj8DiAc5BaU/AiConfig-Editor?type=design&node-id=264-10830&mode=design&t=vIs81NxFIjhv8BbK-0 so we'll want to handle design/style updates after other things are more complete

cc @zakariassoul for visibility here as well